### ISSUE TITLE
BAZEL_USE_CPP_ONLY_TOOLCHAIN usage

### ISSUE THREAD

**Issue by guibou** [tweag/rules_haskell#469]
#453 introduced environment variable `BAZEL_USE_CPP_ONLY_TOOLCHAIN=1` for darwin build on the continuous integration. This is not a satisfying solution because it needs user setup for darwin build and disable objective C build on darwin.

We need at lest:

- [ ] To document this usage in the user guide
- [ ] To get ride of this limitation

**Comment by Profpatsch** [comment]
Any progress on this? I’m not exactly sure what it does, so documentation would be nice.

**Comment by guibou** [comment]
https://github.com/tweag/rules_haskell/pull/453/files#diff-5204cbe23b218cae9de0b1aedd4edcd2R43 explains a bit. I don't know if that's still an issue.

**Comment by Profpatsch** [comment]
> https://github.com/tweag/rules_haskell/pull/453/files#diff-5204cbe23b218cae9de0b1aedd4edcd2R43 explains a bit.

yeah, I don’t see from that description what the problem is. Why do we need to disable xcode?

**Comment by guibou** [comment]
I don't really remember. It was requiring `xcode-select`, which was running `xcrunwrapper.sh` which was dependent of some other stuffs. See for a bit of details: https://github.com/tweag/rules_haskell/issues/461#issuecomment-433363785

**Comment by aherrmann** [comment]
This is still required and documented in the README. https://github.com/tweag/rules_nixpkgs/pull/128 would remove the need to define `BAZEL_USE_CPP_ONLY_TOOLCHAIN`.

**Comment by ylecornec** [comment]
It tried to remove the `BAZEL_USE_CPP_ONLY_TOOLCHAIN` option in 9ddbe98536f0c01f7b04bf095e73af09a1db845e.

But it looks like it is still needed for the case: `bindist (macos)`.
CI fails with: `Error: DEVELOPER_DIR not set`.
